packages/web-api/src/methods.ts

```diff
@@ -1593,25 +1641,174 @@ export interface ChatPostEphemeralArguments extends WebAPICallOptions, TokenOver
+  /**
+   * @description Channel, private group, or IM channel to send message to. Can be an encoded ID, or a name.
+   * @see {@link https://api.slack.com/methods/chat.postMessage#channels `chat.postMessage` Channels reference for more details}.
+   */
   channel: string;
-  text?: string;
-  as_user?: boolean;
-  attachments?: MessageAttachment[];
-  blocks?: (KnownBlock | Block)[];
-  icon_emoji?: string; // if specified, as_user must be false
-  icon_url?: string; // if specified, as_user must be false
-  metadata?: MessageMetadata;
+  /**
+   * @description Find and link user groups. No longer supports linking individual users; use syntax shown in
+   * {@link https://api.slack.com/reference/surfaces/formatting#mentioning-users Mentioning Users} instead.
+   */
   link_names?: boolean;
+  /**
+   * @description Attach metadata to your message. Metadata you post to Slack is accessible to any app or user who is a
+   * member of that workspace.
+   * @see {@link https://api.slack.com/metadata Message Metadata}.
+   */
+  metadata?: MessageMetadata;
+  /**
+   * @description Disable Slack markup parsing by setting to `false`. Enabled by default.
+   */
   mrkdwn?: boolean;
+  /**
+   * @description Change how messages are treated and formatted, depending on the contents of the message and the value
+   * of this property.
+   * @see {@link https://api.slack.com/methods/chat.postMessage#formatting `chat.postMessage` Formatting reference}.
+   */
   parse?: 'full' | 'none';
-  reply_broadcast?: boolean; // if specified, thread_ts must be set
-  thread_ts?: string;
+  /**
+   * @description Pass `true` to enable unfurling of primarily text-based content.
+   */
   unfurl_links?: boolean;
+  /**
+   * @description Pass `false` to disable unfurling of media content.
+   */
   unfurl_media?: boolean;
-  username?: string; // if specified, as_user must be false
 }
-export interface ChatScheduleMessageArguments extends WebAPICallOptions, TokenOverridable {
+
+/**
+ * @description `chat.postMessage` arguments where the `text` property captures the message contents.
+ */
+interface ChatPostMessageArgumentsTextBased extends ChatPostMessageArgumentsBase {
+  /**
+   * @description The formatted text of the message to be published.
+   */
+  text: string;
+  attachments?: never;
+  blocks?: never;
+}
+
+/**
+ * @description `chat.postMessage` arguments where the `attachments` property captures the message contents.
+ */
+interface ChatPostMessageArgumentsAttachmentsBased extends ChatPostMessageArgumentsBase {
+  /**
+   * @description The fallback text used in notifications.
+   */
+  text?: string;
+  attachments: [MessageAttachment, ...MessageAttachment[]];
+}
+
+/**
+ * @description `chat.postMessage` arguments where the `blocks` property captures the message contents.
+ */
+interface ChatPostMessageArgumentsBlocksBased extends ChatPostMessageArgumentsBase {
+  /**
+   * @description The fallback text used in notifications.
+   */
+  text?: string;
```

Comment:
Is this intentionally optional? In my undertstanding, having `text` is still recommended even when having `blocks`.